Create 240314-removing-transfer-transaction-type-SFE.md (#1075)
codatio/codat-docs: blog/240314-removing-transfer-transaction-type-SFE.md

---

> title: "2024-03-14: Removing transfer In/Out transaction type from Sync for Expenses "

Reviewer: ```suggestion
title: "Upcoming 2024-03-14: Removing 'Transfer in/out' transaction types from Sync for Expenses"
```

> date: "2024-03-14"

Reviewer: update the date when finalising the deprecation

tags: ["Deprecation"]
draft: true

> authors: Amy-Roberts

Reviewer: check you have been added to the authors.yaml

---

> On June 19, 2024 we will be removing the Transfer In and Transfer Out transaction type from the Sync for Expenses expense-transaction endpoint. 

Reviewer: also the full name of the expense-transaction endpoint and a link to it please

> To represent a transfer for our expenses offering going forward you will need to use the [transfer-transaction endpoint](link). This endpoint will allow you to represent multi currency transfer, as well as single currency. You can read more information on multicurrency transfers [here](https://docs.codat.io/expenses/fx-management#transfers). 

Reviewer: ```suggestion

After the change, you will need to use the [transfer-transaction - needs the full name of the endpoint please](link) endpoint to represent a transfer when syncing expenses. This endpoint allows you to represent multicurrency and single-currency transfers. You can read more about multicurrency transfers [in our documentation](https://docs.codat.io/expenses/fx-management#transfers). 
```

## Action required

> If you are currently using the Transfer In or Transfer Out transaction type, then you will need to map these transactions to then be created via the create transfer transaction endpoint instead. 

Reviewer: ```suggestion
If you are currently using the `Transfer in` or `Transfer out` transaction type, you will need to create these transactions via the [Create transfer transaction](link) endpoint instead. 
```

> You will also still be able to utilise the same attachment endpoint to push a document alongside your transfer transaction. 

Reviewer: ```suggestion
```

> When a transfer is created via this new endpoint, the response will be a sync id, meaning a sync will be automatically initiated. You will not recieve a dataset id and therefore will not use it to initiate the syncing of transactions.  

Reviewer: ```suggestion
When creating a transfer,  [Create transfer transaction](link) endpoint initiates a sync automatically and will respond with a `syncId`. You do not need to initiate the transaction sync yourself, and you will not receive a `datasetId` in the response.
```

## Expected impact if no action is taken

> If you continue to use the Transfer In/Out transaction type, the syncs fail and will result in the following error message:

Reviewer: ```suggestion
If you try to use the `Transfer in` and `Transfer out` transaction types with the [Full name please expense-transaction](link) endpoint after June 19, 2024, the syncs will fail and return the following error message:
```
Transfer In/Out transaction type is no longer supported. Please use the 'Create transfer transactions' endpoint.
```
```

> ```JSON

Reviewer: so first of all, I don't think you need this - it's enough to list the error message. second, I assume this is a mockup - if not,  then note the typo in the word "Transfer" in the error message
